presto-native-execution/presto_cpp/main/TaskManager.cpp

```diff
@@ -42,30 +42,37 @@ namespace {
 // the file system and sets the path to it to the Task.
 static void maybeSetupTaskSpillDirectory(
     const core::PlanFragment& planFragment,
-    exec::Task& execTask,
+    exec::Task& veloxTask,
     const std::string& baseSpillDirectory) {
   if (baseSpillDirectory.empty() ||
-      !planFragment.canSpill(execTask.queryCtx()->queryConfig())) {
+      !planFragment.canSpill(veloxTask.queryCtx()->queryConfig())) {
     return;
   }
 
   const auto includeNodeInSpillPath =
       SystemConfig::instance()->includeNodeInSpillPath();
   auto nodeConfig = NodeConfig::instance();
-
   const auto taskSpillDirPath = TaskManager::buildTaskSpillDirectoryPath(
       baseSpillDirectory,
       nodeConfig->nodeInternalAddress(),
       nodeConfig->nodeId(),
-      execTask.queryCtx()->queryId(),
-      execTask.taskId(),
+      veloxTask.queryCtx()->queryId(),
+      veloxTask.taskId(),
       includeNodeInSpillPath);
-  execTask.setSpillDirectory(taskSpillDirPath);
-  // Create folder for the task spilling.
-  auto fileSystem =
-      velox::filesystems::getFileSystem(taskSpillDirPath, nullptr);
-  VELOX_CHECK_NOT_NULL(fileSystem, "File System is null!");
-  fileSystem->mkdir(taskSpillDirPath);
+
+  try {
+    // Create folder for the task spilling.
+    auto fileSystem =
+        velox::filesystems::getFileSystem(taskSpillDirPath, nullptr);
+    VELOX_CHECK_NOT_NULL(fileSystem, "File System is null!");
+    fileSystem->mkdir(taskSpillDirPath);
+  } catch (...) {
+    LOG(WARNING) << "Failed to create spill directory for velox task "
+                 << veloxTask.taskId() << ". Error: "
+                 << folly::exceptionStr(std::current_exception());
+    return;
```

Comment:
Why is it OK to ignore this error? If we fail to setup spill directory and the query runs out of memory, will we get some cryptic error after wasting a lot of compute?

Reply:
We currently create spilling directory at the startup of each query so the query might not spill. So instead of failing query, we can proceed the query without spilling enabled. We could later add a thread-safe method at query level to create a spilling directory when needs if this happens too often in practice.

Reply:
I'm not sure I understand. What would cause a failure to setup spill directory? If this is cause by bad config, where base spill directory doesn't exist than it might be better to fail early, no?

Reply:
Not bad config. From the failure that we have been seen in Meta internal testing, it is caused by some remote storage error (some internal timeout at storage backend). The spilling directory setup happens on the query start but not when we trigger spills.